### DISCUSSION TITLE
Teleportation to claims

### DISCUSSION THREAD

**Discussion by TomLewis** [GriefPrevention/GriefPrevention#134] [Ideas]
Ahoy!

A couple of requests from a few years ago I would like to re-submit now we have a new dev :)

- Being able to name claims, not for entry/exit messages but purely for organisation in /claimslist, so players know what claim is what, naming could also be shown when you inspect a claim (An alternative to walking in for a message!) As I know thats been a very requested feature. this could be halfway meeting point and kill two birds with one stone.
- Allow admins to click on claims in /claimslist to teleport to them! This is a bit of a no-brainer for faster admin tools, give it a permission too, maybe show some information when clicking for non-admins. /claimslist is just a mess for anyone to understand at present. this could really be tided up.

I love the layout of LuckPerms /lp user username info, this is so clean and easy to understand. maybe take some inspiration ;p

**Comment by zedwick** [comment]
See #126 for your first request

Would love to see the second 👍

**Comment by RoboMWM** [comment]
`Allow admins to click on claims in /claimslist to teleport to them! `

Will require json. Problem with that is versions could vary in API (afaik, not sure if CB has any json-string stuff) as well as the tellraw (or whichever command is used) parameters.

**Comment by zedwick** [comment]
Spigot has the Spigot Chat API, which allows for fancy hover overs and clickable text. If Craftbukkit support is still required, then falling back to legacy text would be fine on craftbukkit servers.

**Comment by RoboMWM** [comment]
FYI, I made claimlistclassifier, an addon that allows for claim naming and listing. Could maybe also add clickable teleportation to the printed list if requested. https://github.com/RoboMWM/claimlistclassifier

**Comment by TomLewis** [comment]
I'm confused, wouldn't it be easier to keep all requests in one place? this github repo? Both points are simple additions that would benefit everyone using GP.
**Edit:** I just tested this plugin, this is amazing to name claims! that's legit no1 done! It just needs the click to teleport to the claim! I had no idea this existed!

**Comment by RoboMWM** [comment]
Hence why I haven't closed this - but since this is an extraneous feature that requires removing CB support (or at least building off of CB - there are actually a few servers who use CB still, interestingly), it's addon territory.

I may look into baking in claim sorting for version 20, since this is only needed on command. The minor performance hit due to making a copy of the claim and sorting it can be mitigated ~~in an asynchronous task.~~ hmm, might not be safe since I still need to access locations to sort.

**Comment by TomLewis** [comment]
When you say CB are you referencing CraftBukkit? Im surprised anyone uses CraftBukkit in 2018! Im a Paper man myself. 
I would honestly like the option to set the formatting for the claimslist myself, its such a mess and impossible to use at present. I installed the name addon, so thats a good start!

**Comment by RoboMWM** [comment]
Yes, CB = CraftBukkit.

This is possible in the ClaimslistClassifier addon.

**Comment by TomLewis** [comment]
Fantastic!
